packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -1873,6 +1874,16 @@ class FocusManager with DiagnosticableTreeMixin, ChangeNotifier {
     assert(_focusDebug(() => 'Notified ${_dirtyNodes.length} dirty nodes:', () => _dirtyNodes));
     _dirtyNodes.clear();
     if (previousFocus != _primaryFocus) {
+      if (_primaryFocus != null && (_primaryFocus!.context?.mounted ?? false)) {
```

Comment:
So I think we need to store the latest focused flutter view somewhere and use that to compute the delta to send the right request. See https://github.com/flutter/engine/blob/main/lib/web_ui/lib/src/engine/platform_dispatcher/view_focus_binding.dart#L87-L105

Reply:
Since the engine is the one that owns the view focus, shouldn't it use its idea of the last view focused (i.e. `_lastViewId` in that code snippet)?

Reply:
> Since the engine is the one that owns the view focus, shouldn't it use its idea of the last view focused (i.e. _lastViewId in that code snippet)?

I do get your point. Basically making requestViewFocus idempotent, am I right? Makes sense; but I do think we should make these calls *only* when they're needed. Am I right in assuming that in a single view world, `requestViewFocusChange` will be called as many times (with the same params) as focus traverseses among "widgets"?

Reply:
> Am I right in assuming that in a single view world, requestViewFocusChange will be called as many times (with the same params) as focus traverses among "widgets"?

Yes, that's true.

I wonder if the `FlutterView` class should keep track of whether it is focused or not?  If it did, then I could ask it if it was currently focused, and only call `requestViewFocusChange` if it wasn't focused already.

Reply:
Okay, check out this PR and see if it seems like a good idea to you: https://github.com/flutter/engine/pull/53723